cmake_test/asserts/library_exists.cmake

```diff
@@ -0,0 +1,41 @@
+include_guard()
+
+#[[[ Asserts that a library exists.
+#
+# Asserts that the target exists and is a library.
+#
+# :param _ale_name: The name of the library
+# :type _afe_path: String
+#]]
+function(ct_assert_library_exists _ale_name)
```

Comment:
I'd change the name of this assert as I find it misleading. To me the library is the `.so` or `.a` binary file on disk. This assert is checking that a target defines a library.